### PR TITLE
Add AMI builds to Drone

### PR DESCRIPTION
**Pull request by webvictim**
This PR adds AMI builds to Drone, built the same way Jenkins currently does.

It also:
- Renames `assets/marketplace` to `assets/aws` to more accurately reflect the data contained there, and updates all references to that path in the repo which I could find
  - This will require an update to the two Jenkins AMI build jobs to look for their `Jenkinsfile`s in this new location - I will do this once the PR is merged.
- Adds support to `files/install.sh` for using a tarball pulled from the staging S3 bucket, rather than downloading from get.gravitational.com
  - This allows easier testing of the AMI build flow via PR if needed. It also means that we can build the AMIs at the same time as we push a tag, rather than needing to wait for a promotion job.
- Renames the Makefile target `oss-jenkins-build` to `oss-ci-build`
- Renames the Makefile target `ent-jenkins-build` to `ent-ci-build`
- Changes "Jenkins" references to "CI"

It also fixes some Cloudformation links in the docs that were pointing to a removed directory.

Relevant to #3936

### CODE REVIEW

**Comment by webvictim** [comment]
retest this please

**Review by awly — APPROVED** [review]

**Comment by webvictim** [comment]
@benarent Please take a look at the `.md` files as I need code owner approval from you :)

File: docs/4.0/aws_oss_guide.md
Comment (benarent) [inline docs/4.0/aws_oss_guide.md:302]:
```suggestion
[AWS AMI example](https://github.com/gravitational/teleport/tree/master/examples/aws/cloudformation#aws-cloudformation-based-provisioning-example). It requires a VPC, but
```

File: docs/4.1/aws_oss_guide.md
Comment (benarent) [inline docs/4.1/aws_oss_guide.md:356]:
```suggestion
[AWS AMI example](https://github.com/gravitational/teleport/tree/master/examples/aws/cloudformation#aws-cloudformation-based-provisioning-example). It requires a VPC, but
```

File: docs/4.2/aws_oss_guide.md
Comment (benarent) [inline docs/4.2/aws_oss_guide.md:364]:
```suggestion
[AWS AMI example](https://github.com/gravitational/teleport/tree/master/examples/aws/cloudformation#aws-cloudformation-based-provisioning-example). It requires a VPC, but
```

File: docs/4.3/aws_oss_guide.md
Comment (benarent) [inline docs/4.3/aws_oss_guide.md:356]:
```suggestion
[AWS AMI example](https://github.com/gravitational/teleport/tree/master/examples/aws/cloudformation#aws-cloudformation-based-provisioning-example). It requires a VPC, but
```

**Review by benarent — APPROVED** [review]
Added a few suggestions.

**Comment by webvictim** [comment]
retest this please

**Pull request merged**
